samcli/lib/bootstrap/nested_stack/nested_stack_builder.py

```diff
@@ -43,11 +45,13 @@ def add_function(
     @staticmethod
     def get_layer_logical_id(function_logical_id: str) -> str:
         function_logical_id_hash = str_checksum(function_logical_id)
+        function_logical_id = NestedStackBuilder._get_logical_id_compliant_str(function_logical_id)
         return f"{function_logical_id[:48]}{function_logical_id_hash[:8]}DepLayer"
 
     @staticmethod
     def get_layer_name(stack_name: str, function_logical_id: str) -> str:
         function_logical_id_hash = str_checksum(function_logical_id)
+        function_logical_id = NestedStackBuilder._get_logical_id_compliant_str(function_logical_id)
```

Comment:
nit: Can we rename this variable here and above to something more specific to describe that it's been mutated?

Reply:
Good suggestion, addressed in the latest commit 👍🏼